core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/RCodegen.scala

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.codegen
+
+import com.microsoft.azure.synapse.ml.build.BuildInfo
+import com.microsoft.azure.synapse.ml.codegen.CodegenConfigProtocol._
+import com.microsoft.azure.synapse.ml.core.env.FileUtilities._
+import com.microsoft.azure.synapse.ml.core.utils.JarLoadingUtils.instantiateServices
+import org.apache.commons.io.FileUtils
+import spray.json._
+
+import java.io.File
+
+object RCodegen {
+
+  import CodeGenUtils._
+
+  def generateRClasses(conf: CodegenConfig): Unit = {
+    val instantiatedClasses = instantiateServices[RWrappable](conf.jarName)
+    instantiatedClasses
+      .foreach { w =>
+        println(w.getClass.getName)
+        w.makeRFile(conf)
+    }
+  }
+
+  //noinspection ScalaStyle
+  def generateRPackageData(conf: CodegenConfig): Unit = {
```

Comment:
This is a separate object but python and (maybe dotnet?) are not. We should have a consistent set of abstractions here

Reply:
DotnetCodegen is a separate object and I chose to follow that model. I will look at making python conform.